flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializer.java

```diff
@@ -795,13 +782,15 @@ public PojoSerializerConfigSnapshot(
 			this.nonRegisteredSubclassesToSerializerConfigSnapshots =
 					Preconditions.checkNotNull(nonRegisteredSubclassesToSerializerConfigSnapshots);
 
-			this.ignoreTypeSerializerSerialization = ignoreTypeSerializerSerialization;
+			this.excludeSerializers = excludeSerializers;
 		}
 
 		@Override
 		public void write(DataOutputView out) throws IOException {
 			super.write(out);
 
+			out.writeBoolean(excludeSerializers);
+
 			// --- write fields and their serializers, in order
 
 			out.writeInt(fieldToSerializerConfigSnapshot.size());
```

Comment:
What I just noticed is that this is almost 3x duplicated code. I suggest to create a private method that writes `Map<Class<?>, Tuple2<TypeSerializer<?>, TypeSerializerConfigSnapshot>>` and call it 3 times.